diff --git a/RestrictEvents/RestrictEvents.cpp b/RestrictEvents/RestrictEvents.cpp
index 27562f0..66365a3 100644
--- a/RestrictEvents/RestrictEvents.cpp
+++ b/RestrictEvents/RestrictEvents.cpp
@@ -61,6 +61,7 @@ struct RestrictEventsPolicy {
 		static const char *procBlacklist[] {
 			"/System/Library/CoreServices/ExpansionSlotNotification",
 			"/System/Library/CoreServices/MemorySlotNotification",
+			"/usr/libexec/displaypolicyd",
 		};
 
 		char pathbuf[MAXPATHLEN];
